Cloudflare projects require cloudflare vite-plugin

I noticed that v1.0.0-alpha.7 broke the ability to deploy the default template to cloudflare.

```
npm create waku
...
CLOUDFLARE=1 npm run build
npx wrangler deploy
``` 

It works with `npm create waku@0.12.5-1.0.0-alpha.6-0` and breaks with 0.12.5-1.0.0-alpha.7-0.

The error looks like this -

✘ [ERROR] A request to the Cloudflare API (/accounts/.../workers/scripts/waku-project/versions) failed.

  Uncaught Error: No such module "node:module".
    imported from "assets/main.react-server-B1jRPgjI.js"
   [code: 10021]
  To learn more about this error, visit:
  https://developers.cloudflare.com/workers/observability/errors/#validation-errors-10021

It seems to come from node:module being imported at the top-level. `alpha.6` only ouputs node:module in dist/server/build.js but `alpha.7` outputs `import "node:module";` in dist/server/assets/main.react-server-CXiOEMoQ.js. I'm not sure what it is even doing there. It's the first line in that file. If I delete it, I'm able to deploy to cloudflare and it seems that everything works. We could work around it by deleting that line in a post-build step.

If I add @cloudflare/vite-plugin to the waku.config.ts vite plugins, it also deploys without any issues. There is no line in the build output with `import "node:module";` (which I'm checking for by using `grep -r "node:module" dist` after building).

--- a/e2e/multi-platform.spec.ts
+++ b/e2e/multi-platform.spec.ts
@@ -3,6 +3,7 @@ import {
   cpSync,
   existsSync,
   readFileSync,
+  readdirSync,
   rmSync,
   symlinkSync,
   writeFileSync,
@@ -36,32 +37,41 @@ const waku = fileURLToPath(
 type BuildPlatformTarget = {
   adapter: string;
   clearDirOrFile: string[];
-  checkJsonFile?: (dir: string) => boolean;
+  assertBuildOutput?: (dir: string) => void;
 };
 
-const hasDistServerWranglerMainIndexJs = (dir: string) => {
-  const file = join(dir, 'dist', 'server', 'wrangler.json');
-  if (!existsSync(file)) {
-    return false;
+const directoryContainsText = (dir: string, text: string): boolean => {
+  for (const entry of readdirSync(dir, { withFileTypes: true })) {
+    const file = join(dir, entry.name);
+    if (entry.isDirectory()) {
+      if (directoryContainsText(file, text)) {
+        return true;
+      }
+    } else if (entry.isFile() && readFileSync(file, 'utf-8').includes(text)) {
+      return true;
+    }
   }
+  return false;
+};
+
+const assertCloudflareBuildOutput = (dir: string) => {
+  const file = join(dir, 'dist', 'server', 'wrangler.json');
+  expect(existsSync(file)).toBe(true);
   const json = JSON.parse(readFileSync(file, 'utf-8')) as {
     main?: string;
     assets?: { directory?: string };
   };
-  if (json.main !== 'index.js') {
-    return false;
-  }
-  if (!json.assets?.directory) {
-    return false;
-  }
+  expect(json.main).toBe('index.js');
+  expect(json.assets?.directory).toBeTruthy();
   const deployConfig = join(dir, '.wrangler', 'deploy', 'config.json');
-  if (!existsSync(deployConfig)) {
-    return false;
-  }
+  expect(existsSync(deployConfig)).toBe(true);
   const deployJson = JSON.parse(readFileSync(deployConfig, 'utf-8')) as {
     configPath?: string;
   };
-  return deployJson.configPath === '../../dist/server/wrangler.json';
+  expect(deployJson.configPath).toBe('../../dist/server/wrangler.json');
+  expect(
+    directoryContainsText(join(dir, 'dist', 'server'), 'node:module'),
+  ).toBe(false);
 };
 
 const buildPlatformTarget: BuildPlatformTarget[] = [
@@ -76,7 +86,7 @@ const buildPlatformTarget: BuildPlatformTarget[] = [
   {
     adapter: 'cloudflare',
     clearDirOrFile: ['dist', 'wrangler.jsonc', '.wrangler'],
-    checkJsonFile: hasDistServerWranglerMainIndexJs,
+    assertBuildOutput: assertCloudflareBuildOutput,
   },
   {
     adapter: 'deno',
@@ -114,7 +124,7 @@ test.describe(`multi platform builds`, () => {
     for (const {
       adapter,
       clearDirOrFile,
-      checkJsonFile = () => true,
+      assertBuildOutput,
     } of buildPlatformTarget) {
       test(`build ${project} with ${adapter} should not throw error`, async () => {
         const temp = makeTempDir(project);
@@ -137,7 +147,7 @@ test.describe(`multi platform builds`, () => {
               env: process.env,
             }),
           ).resolves.not.toThrow();
-          expect(checkJsonFile(temp)).toBe(true);
+          assertBuildOutput?.(temp);
         } finally {
           rmSync(temp, { recursive: true, force: true, maxRetries: 3 });
         }

--- a/e2e/multi-platform.spec.ts
+++ b/e2e/multi-platform.spec.ts
@@ -34,20 +34,23 @@ const waku = fileURLToPath(
   new URL('../packages/waku/dist/cli.js', import.meta.url),
 );
 
+const STATIC_NODE_MODULE_IMPORT_RE =
+  /(?:^|[;\n])\s*import(?:\s+[\w*\s{},]+?\s+from)?\s*['"]node:module['"]/m;
+
 type BuildPlatformTarget = {
   adapter: string;
   clearDirOrFile: string[];
   assertBuildOutput?: (dir: string) => void;
 };
 
-const directoryContainsText = (dir: string, text: string): boolean => {
+const directoryContainsPattern = (dir: string, pattern: RegExp): boolean => {
   for (const entry of readdirSync(dir, { withFileTypes: true })) {
     const file = join(dir, entry.name);
     if (entry.isDirectory()) {
-      if (directoryContainsText(file, text)) {
+      if (directoryContainsPattern(file, pattern)) {
         return true;
       }
-    } else if (entry.isFile() && readFileSync(file, 'utf-8').includes(text)) {
+    } else if (entry.isFile() && pattern.test(readFileSync(file, 'utf-8'))) {
       return true;
     }
   }
@@ -70,7 +73,10 @@ const assertCloudflareBuildOutput = (dir: string) => {
   };
   expect(deployJson.configPath).toBe('../../dist/server/wrangler.json');
   expect(
-    directoryContainsText(join(dir, 'dist', 'server'), 'node:module'),
+    directoryContainsPattern(
+      join(dir, 'dist', 'server'),
+      STATIC_NODE_MODULE_IMPORT_RE,
+    ),
   ).toBe(false);
 };
 

--- a/packages/waku/src/lib/vite-entries/entry.build.ts
+++ b/packages/waku/src/lib/vite-entries/entry.build.ts
@@ -1,4 +1,3 @@
-import { createRequire } from 'node:module';
 import { pathToFileURL } from 'node:url';
 import { buildMetadata } from 'virtual:vite-rsc-waku/build-metadata';
 import serverEntry from 'virtual:vite-rsc-waku/server-entry';
@@ -8,7 +7,7 @@ import type { Unstable_EmitFile } from '../types.js';
 import { joinPath } from '../utils/path.js';
 import { stringToStream } from '../utils/stream.js';
 
-function resolveModuleId(moduleId: string, rootDir: string) {
+async function resolveModuleId(moduleId: string, rootDir: string) {
   if (moduleId.startsWith('file://')) {
     return moduleId;
   }
@@ -16,6 +15,7 @@ function resolveModuleId(moduleId: string, rootDir: string) {
     // treat as project-root relative (not filesystem root)
     return pathToFileURL(joinPath(rootDir, moduleId.slice(1))).href;
   }
+  const { createRequire } = await import('node:module');
   const require = createRequire(joinPath(rootDir, 'DUMMY.js'));
   const resolved = require.resolve(moduleId);
   return pathToFileURL(resolved).href;
@@ -32,7 +32,7 @@ export async function INTERNAL_runBuild({
   const prunableFiles = new Set<string>();
   let build = serverEntry.build;
   for (const enhancer of serverEntry.buildEnhancers || []) {
-    const moduleId = resolveModuleId(enhancer, rootDir);
+    const moduleId = await resolveModuleId(enhancer, rootDir);
     const mod = await import(moduleId);
     build = await mod.default(build);
   }

